tower-buffer/src/worker.rs

```diff
@@ -157,30 +165,43 @@ where
 
         loop {
             match try_ready!(self.poll_next_msg()) {
-                Some(msg) => {
+                Some((msg, first)) => {
+                    let _guard = msg.span.enter();
                     if let Some(ref failed) = self.failed {
+                        tracing::trace!("notifying about worker failure");
```

Comment:
Nit: notifying _what_ about worker failure? Maybe
```suggestion
                        tracing::trace!("notifying caller about worker failure");
```
or something?